### PR TITLE
Remove aggregate step

### LINKED ISSUE #5716: Remove the `aggregate` step
## Summary

The `in_parallel` step was introduced a long while back as an alternative to the `aggregate` step, which has been around since pre-1.0 days. We added a deprecation notice and an EOL timeline of a year per https://github.com/concourse/concourse/issues/4379#issuecomment-528521450 - and it's been a year, so in v7.0 we can finally remove it.

## Context

* Prior discussion: https://github.com/concourse/concourse/issues/4379#issuecomment-528521450

### PR DESCRIPTION
**Pull request by taylorsilva**
## What does this PR accomplish?
Bug Fix | Feature | Documentation

closes #5716 

## Changes proposed by this PR:
Remove the `aggregate` step.

## Release Note
* Removing the `aggregate` step as planned. It is succeeded by the `in_parallel` step.

## Contributor Checklist
- [x] Followed [Code of conduct], [Contributing Guide] & avoided [Anti-patterns]
- [x] [Signed] all commits
- [x] Added tests (Unit and/or Integration)
- [x] Updated [Documentation]
- [x] Added release note (Optional)

[Code of Conduct]: https://github.com/concourse/concourse/blob/master/CODE_OF_CONDUCT.md
[Contributing Guide]: https://github.com/concourse/concourse/blob/master/CONTRIBUTING.md
[Anti-patterns]: https://github.com/concourse/concourse/wiki/Anti-Patterns
[Signed]: https://help.github.com/en/github/authenticating-to-github/signing-commits
[Documentation]: https://github.com/concourse/docs

## Reviewer Checklist

- [ ] Code reviewed
- [ ] Tests reviewed
- [ ] Documentation reviewed
- [ ] Release notes reviewed
- [ ] PR acceptance performed
- [ ] New config flags added? Ensure that they are added to the
  [BOSH](https://github.com/concourse/concourse-bosh-release) and
  [Helm](https://github.com/concourse/helm) packaging; otherwise, ignored for
  the [integration
  tests](https://github.com/concourse/ci/tree/master/tasks/scripts/check-distribution-env)
  (for example, if they are Garden configs that are not displayed in the
  `--help` text).

### CODE REVIEW

File: atc/public_plan_test.go [hunk 1]
```diff
@@ -401,10 +405,10 @@ var _ = Describe("Plan", func() {
 			Expect(json).ToNot(BeNil())
 			Expect([]byte(*json)).To(MatchJSON(`{
   "id": "0",
-  "aggregate": [
+  "in_parallel": [
```
Comment (aoldershaw) [inline atc/public_plan_test.go:408]: looks like the tests are failing here. should this be:

```
"in_parallel": {
  "steps": [
    ...
  ]
}
```

**Comment by aoldershaw** [comment]
Actually, looks like what I said in standup was wrong - without the 2nd commit, we're still able to view old build logs containing `aggregate` in the UI, since the public plan is already generated and is just treated as raw JSON when sending to the client.

Not sure if it's worth keeping around, but it's a pretty small amount of code. Would be nice to get rid of everything though. What do you think?

**Comment by taylorsilva** [comment]
Yay! Glad old build plans will still render! With that in mind I'm 100% for getting rid of all the code then. It is a small amount of code but I can image the confusion someone may have looking at the elm code, seeing references to `aggregate` and not seeing `aggregate` anywhere in the server-side code. It doesn't help anyone in the future keeping that code around. 🪓

**Comment by aoldershaw** [comment]
> Glad old build plans will still render! With that in mind I'm 100% for getting rid of all the code then

I may not have been totally clear - I meant that the build plans will only render if we *don't* get rid of all the code. With that said, I agree that leaving the Elm code adds more confusion, so I'm down to get rid of it

**Review by aoldershaw — APPROVED** [review]

**Pull request merged**